packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -489,13 +543,90 @@ class SliderThemeData extends Diagnosticable {
     properties.add(DiagnosticsProperty<Color>('disabledThumbColor', disabledThumbColor, defaultValue: defaultData.disabledThumbColor, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<Color>('overlayColor', overlayColor, defaultValue: defaultData.overlayColor, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<Color>('valueIndicatorColor', valueIndicatorColor, defaultValue: defaultData.valueIndicatorColor));
+    properties.add(DiagnosticsProperty<SliderTrackShape>('trackShape', trackShape, defaultValue: defaultData.trackShape, level: DiagnosticLevel.debug));
+    properties.add(DiagnosticsProperty<SliderTickShape>('tickShape', tickShape, defaultValue: defaultData.tickShape, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<SliderComponentShape>('thumbShape', thumbShape, defaultValue: defaultData.thumbShape, level: DiagnosticLevel.debug));
+    properties.add(DiagnosticsProperty<SliderComponentShape>('overlayShape', overlayShape, defaultValue: defaultData.overlayShape, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<SliderComponentShape>('valueIndicatorShape', valueIndicatorShape, defaultValue: defaultData.valueIndicatorShape, level: DiagnosticLevel.debug));
     properties.add(EnumProperty<ShowValueIndicator>('showValueIndicator', showValueIndicator, defaultValue: defaultData.showValueIndicator));
     properties.add(DiagnosticsProperty<TextStyle>('valueIndicatorTextStyle', valueIndicatorTextStyle, defaultValue: defaultData.valueIndicatorTextStyle));
   }
 }
 
+/// Base class for slider track shapes.
+///
+/// Create a subclass of this if you would like a custom track shape.
+///
+/// See also:
+///
+///  * [RectangularSliderTrackShape] for the default track shape.
+abstract class SliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
```

Comment:
Nit: First sentence of dartdoc goes on it's own line. Alternatively you could change the . to a , and accomplish the same thing. 

https://www.dartlang.org/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

Reply:
I am now wondering the comments on the abstract const constructors are useful in this class. Isn't this sort of thing implied by the language and the fact that its an abstract class constructor already?

Reply:
Actually, I did make an issue: https://github.com/flutter/flutter/issues/25672

for this. Since this was copied from the current pattern, I am going to leave it as is, then update the docs in a follow up PR to make them consistent